test/Microsoft.AspNetCore.Mvc.Razor.Test/Internal/ViewComponentTagHelperDescriptorFactoryTest.cs

```diff
@@ -87,6 +86,13 @@ public static TheoryData TypeData
         [MemberData(nameof(TypeData))]
         public void GetCSharpTypeName_ReturnsCorrectTypeNames(Type type, string expected)
         {
+            // Arrange
+            // Work around problems serializing / deserializing this ref type.
+            if (type == null)
+            {
+                type = typeof(ViewComponentTagHelperDescriptorFactoryTest).GetMethod("MethodWithOutParam").GetParameters().First().ParameterType;
+            }
```

Comment:
Not yet. I was waiting for your PR to repro this issue. I'll link it here once I am done.